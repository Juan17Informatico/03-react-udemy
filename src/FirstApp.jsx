import PropTypes from "prop-types";
// const getSaludo = (name) => {
//     return "Hola " + name;
// };

export const FirstApp = ({ title, subTitle, name }) => {
    // console.log(props);

    return (
        <>
            <h1 data-testid="test-title"> { title } </h1>
            <h2>{subTitle}</h2>
            <h2>{subTitle}</h2>
            <h3>{name}</h3>
        </>
    );
};

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
    // title: "No hay titulo",
    subTitle: "No hay Subtitulo",
    name: "Juan Campuzano",
};
