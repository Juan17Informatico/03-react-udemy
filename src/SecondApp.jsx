
export const SecondApp = ()=> {
    return (
        <h1>Second APP</h1>
    );
}