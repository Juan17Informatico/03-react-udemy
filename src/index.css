html,
body {
    background-color: #1a0d01;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}
