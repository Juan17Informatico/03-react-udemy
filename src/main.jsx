import React from "react";
import ReactDOM from "react-dom/client";
import { FirstApp } from "./FirstApp";
// import { CounterApp } from "./CounterApp";
// import { SecondApp } from "./SecondApp";
import "./index.css";


ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        {/* <CounterApp value={10}/> */}
        <FirstApp title="Hola, No soy Nadie, Soy Vegeta" />
    </React.StrictMode>
);

