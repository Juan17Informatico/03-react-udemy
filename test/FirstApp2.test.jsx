/** @jest-environment jsdom */
import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en <FirstApp />", () => {
    const title = "Hola, Soy Goku";

    const subTitle = "Soy un subtitulo"; 

    test("Debe hacer match con el snapshot", () => {
        const { container } = render(<FirstApp title={title}></FirstApp>);

        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar el mensaje "Hola Soy Goku" ', () => {
        render(<FirstApp title={title}></FirstApp>);
        // screen.debug();

        expect(screen.getByText(title)).toBeTruthy();
    });

    test("debe de mostrar el titulo en un h1", () => {
        render(<FirstApp title={title}></FirstApp>);

        expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(title);
    });

    test("debe de mostrar el subitrutlo enviado por props", () => {
        
        render(<FirstApp title={title} subTitle={subTitle}></FirstApp>);

        expect(screen.getAllByText(subTitle).length).toBe(2);

    });
});
