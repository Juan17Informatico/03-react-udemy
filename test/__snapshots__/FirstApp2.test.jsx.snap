// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, Soy Goku
     
  </h1>
  <h2>
    No hay Subtitulo
  </h2>
  <h2>
    No hay Subtitulo
  </h2>
  <h3>
    Juan Campuzano
  </h3>
</div>
`;
